src/main/java/org/casbin/jcasbin/util/BuiltInFunctions.java

```diff
@@ -189,4 +203,61 @@ public String getName() {
             }
         };
     }
+
+    /**
+     * eval calculates the stringified boolean expression and return its result.
+     * The syntax of expressions is exactly the same as Java.
+     * Flaw: dynamically generated classes or non-static inner class cannot be used.
+     * @author tldyl
+     * @since 2020-07-02
+     *
+     * @param eval Boolean expression.
+     * @param env Parameters.
+     * @return The result of the eval.
+     */
+    public static boolean eval(String eval, Map<String, Object> env) {
+        Map<String, EvalModel> evalModels = getEvalModels(env);
+        try {
+            for (String key : evalModels.keySet()) {
+                interpreter.set(key, evalModels.get(key));
+            }
+            return (boolean) interpreter.eval(eval);
+        } catch (EvalError evalError) {
+            evalError.printStackTrace();
+        }
+        return false;
+    }
+
+    /**
+     * getEvalModels extracts the value from env and assemble it into a EvalModel object.
+     *
+     * @param env the map.
+     */
+    private static Map<String, EvalModel> getEvalModels(Map<String, Object> env) {
+        Map<String, EvalModel> evalModels = new HashMap<>();
+        for (String key : env.keySet()) {
+            String[] names = key.split("_");
+            if (!evalModels.containsKey(names[0])) {
+                evalModels.put(names[0], new EvalModel());
+            }
+            switch (names[1]) {
+                case "sub":
+                    evalModels.get(names[0]).sub = env.get(key);
+                    break;
+                case "obj":
+                    evalModels.get(names[0]).obj = env.get(key);
+                    break;
+                case "act":
+                    evalModels.get(names[0]).act = env.get(key);
+                    break;
+            }
+        }
+        return evalModels;
+    }
+
+    public static class EvalModel { //This class must be public and static.
```

Comment:
Why only these 3 elements here? What if we have another element like `dom` or even a customized element defined by user?